### PR TITLE
planner: consider path is global index path or not in `skylinePruning` function

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: ref #52061

Problem Summary: Because we didn't consider `mergeSort` cost for many executors and it's not easy to fix it right now, see [comments](https://github.com/pingcap/tidb/issues/52061#issuecomment-2362859572). So we decided to consider a path is global index or not in `skylinePruning` function.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by qw4990 — APPROVED** [review]

File: pkg/planner/core/logical_plans_test.go
Comment (hawkingrei) [inline pkg/planner/core/logical_plans_test.go:1960]: add ```require.NoError(t, failpoint.Enable(xxx))``` to avoid deleted failpoint.

File: pkg/planner/core/mock.go
Comment (time-and-fate) [inline pkg/planner/core/mock.go:610]:
```suggestion
	// add a global index `b_global` and `b_c_global` and normal index `b` and `b_c`
```

**Review by time-and-fate — APPROVED** [review]
/hold
Others LGTM, feel free to /unhold if you are ready to merge it.

File: pkg/planner/core/logical_plans_test.go
Comment (time-and-fate) [inline pkg/planner/core/logical_plans_test.go:2039]: Note that `c_d_e` actually is not pruned by "skyline pruning".

**Pull request merged**
